python/lsst/sims/utils/AstrometryUtils.py

```diff
@@ -475,27 +475,37 @@ def _appGeoFromICRS(ra, dec, pm_ra=None, pm_dec=None, parallax=None,
     if mjd is None:
         raise RuntimeError("cannot call appGeoFromICRS; mjd is None")
 
-    if isinstance(ra, np.ndarray):
-        fill_value = np.zeros(len(ra))
-    else:
-        fill_value = 0.0
+    include_px = False
+
+    if (pm_ra is not None or pm_dec is not None or
+        v_rad is not None or parallax is not None):
+
+        include_px = True
+
+        if isinstance(ra, np.ndarray):
+            fill_value = np.zeros(len(ra))
```

Comment:
Should this have a `dtype=float`? I think it might end up as an `int` since `len` returns an int.